File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        result = 0
        
        row = 0
        col = 0
        
        while row < len(mat) and col < len(mat):
            result += mat[row][col]
            result += mat[len(mat)-1-row][col]
            
            row += 1
            col += 1
        
        if len(mat) % 2 == 1:
            mid = len(mat) // 2
            result -= mat[mid][mid]
        
        return result

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.swift
class Solution {
    func diagonalSum(_ mat: [[Int]]) -> Int {
        var result = 0
        
        var row = 0
        var col = 0
        
        while row < mat.count && col < mat.count {
            result += mat[row][col]
            result += mat[(mat.count - row - 1)][col]
            
            row += 1
            col += 1
        }
        
        if mat.count % 2 == 1 {
            let mid = mat.count / 2
            result -= mat[mid][mid]
        }
        
        return result
    }
}